mlir/test/Conversion/VectorToLLVM/vector-to-llvm.mlir

```diff
@@ -187,7 +187,7 @@ func.func @broadcast_vec2d_from_vec0d(%arg0: vector<f32>) -> vector<3x2xf32> {
 //       CHECK: %[[T0:.*]] = builtin.unrealized_conversion_cast %[[A]] : vector<f32> to vector<1xf32>
 //       CHECK: %[[T1:.*]] = ub.poison : vector<3x2xf32>
 //       CHECK: %[[T2:.*]] = builtin.unrealized_conversion_cast %[[T1]] : vector<3x2xf32> to !llvm.array<3 x vector<2xf32>>
-//       CHECK: %[[T4:.*]] = llvm.mlir.constant(0 : index) : i64
+//       CHECK: %[[T4:.*]] = llvm.mlir.constant(0 : i64) : i64
```

Comment:
Just double-checking, I don't see anything in this PR that would change this type?

In practice, `index`, IIRC, will be lowered to either `i64` or `i32`, but that's not guaranteed. There's also a PR to extend the related logic: https://github.com/llvm/llvm-project/pull/115808

Now, this PR is somewhat tangential to the lowering of `index`. Just pointing out that this is likely a corner case that might require re-visiting.

Reply:
This is not because of the index type, but the lowering difference between vector.extract and vector.extractelement. vector.extract lowering to llvm is better, and actually uses the type converter on index while lowering. vector.extractelement doesn't and is a worse lowering. This is just a side effect of vector.extract lowering being better.